app/controllers/filler/rarePokemon.js

```diff
@@ -1,107 +1,65 @@
-        fs.readdir(dirPath, function(err, filenames) {
+        // recursively calling the function for continuius listening
+        setTimeout(function(){
+            module.exports.insertToDb();
+        },  config.pokesniper.listeningInterval);
+    },
 
-            let noOfFiles = filenames.length,
-                count =0;
 
-            if (err) {
-                logger.error('Error in reading directory' + dirPath);
-                process.exit();
-            }
-            // when no files are present then exit the process
-            if( noOfFiles === 0){
-                logger.info('No files to read');
-                process.exit();
+    /*
+     * inserting the read data into MongoDB
+     */
+    addRarePokemon : function (rarePokemon) {
+        rarePokemonStore.add(rarePokemon, function (success, data) {
+            if (success === 1) {
+                logger.success('Success: ' + data);
+            } else {
+                logger.error('Error:' + data);
             }
-
-            logger.info('MongoDb Insertion...');
-
-            // reading individual files for the data
-            filenames.forEach(function(filename) {
-                fs.readFile(dirPath + filename, 'utf-8', function(err, rarePokemons) {
-
-                    logger.info('filename', filename);
-                    logger.info('\n');
-
-                    // some error on reading the file
-                    if (err) {
-                        logger.error('Error in reading file' + filename);
-                        return;
-                    }
-
-                    if (rarePokemons !== undefined) {
-
-                        // inserting the read content into MongoDB
-                        insert(rarePokemons, function(success){
-                            //decrease the number of files to be read
-                            noOfFiles--;
-                            // deleting the file after it has been read and data being stored into database
-                            fs.unlinkSync(dirPath + filename);
-                            // if no files then sent a callback to calling function
-                            if(noOfFiles === 0)
-                                callback(true);
-                        });
-                    }
-
-
-
-                });
-            });
         });
+    },
 
-        /*
-         * inserting the read data into MongoDB
-         */
-        let addRarePokemon = function (rarePokemon, callback) {
-            rarePokemonStore.add(rarePokemon, function (success, data) {
-                callback(success);
+/*
+ * inserting the read data into MongoDB
+ */
+    insert : function (rarePokemons) {
+        let length = 0,
+            pokemonName;
+
+        async.forEach(rarePokemons, function(rarePokemon){
+            length++;
+            pokemonName = (rarePokemons[0].name);
+            // extracting pokemon id from the names
```

Comment:
Seems like this should be rarePokemon.name. 
rarePokemons[0].name will always consider the first object, which is not the expected behavior if you have a loop.